evalml/pipelines/components/transformers/preprocessing/text_featurizer.py

```diff
@@ -126,16 +130,48 @@ def transform(self, X, y=None):
         if X_nlp_primitives.isnull().any().any():
             X_nlp_primitives.fillna(0, inplace=True)
 
-        X_lsa = self._lsa.transform(X_ww.ww[self._text_columns])
+        X_ww_altered = infer_feature_types(
```

Comment:
@bchen1116 I think we can simplify this implementation if we keep a mask of the rows that were NaN. I'm posting it here but down to sit down and talk through it together as well!

```python
    def transform(self, X, y=None):
        """Transforms data X by creating new features using existing text columns

        Arguments:
            X (pd.DataFrame): The data to transform.
            y (pd.Series, optional): Ignored.

        Returns:
            pd.DataFrame: Transformed X
        """
        X_ww = infer_feature_types(X)
        if self._features is None or len(self._features) == 0:
            return X_ww

        es = self._make_entity_set(X_ww, self._text_columns)
        nan_mask = X[self._text_columns].isna()
        any_nans = nan_mask.any().any()
        X_nlp_primitives = ft.calculate_feature_matrix(
            features=self._features, entityset=es
        )
        if X_nlp_primitives.isnull().any().any():
            X_nlp_primitives.fillna(0, inplace=True)

        if any_nans:
            primitive_features = self._get_primitives_provenance(self._features)
            for column, derived_features in primitive_features.items():
                X_nlp_primitives.loc[nan_mask[column], derived_features] = pd.NA

        X_ww_altered = infer_feature_types(
            X_ww.ww[self._text_columns].fillna(""),
            {s: "NaturalLanguage" for s in self._text_columns},
        )
        X_lsa = self._lsa.transform(X_ww_altered)
        if any_nans:
            lsa_features = self._lsa._get_feature_provenance()
            for column, derived_features in lsa_features.items():
                X_lsa.loc[nan_mask[column], derived_features] = pd.NA

        X_nlp_primitives.set_index(X_ww.index, inplace=True)

        X_ww = X_ww.ww.drop(self._text_columns)
        for col in X_nlp_primitives:
            X_ww.ww[col] = X_nlp_primitives[col]
        for col in X_lsa:
            X_ww.ww[col] = X_lsa[col]
        return X_ww
```

I think it's simpler because I think it's easier to understand what's happening (no regex, fewer lines of code) and it uses abstractions we already have in place (the feature provenance). It also doesn't require us to call `infer_feature_types` at the end - I'm afraid that may cause a bug down the line? At least that's what happened in https://github.com/alteryx/evalml/pull/2297 .

It may also be more performant? Haven't quantified this but with this implementation there's less space used (yes we keep a mask of NaNs but there are two calls to copy in `_empty_to_nan`) and the operation of replacing values in place is vectorized whereas calling `df.apply` is not.

Reply:
@freddyaboulton I'm disappointed you didn't submit a PR to the PR.

Reply:
🙈

Reply:
@freddyaboulton thanks for the suggestion! Yep, can definitely use this!